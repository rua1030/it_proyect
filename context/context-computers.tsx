'use client';

import React, {
  createContext, useState, useContext, ReactNode,
} from 'react';

const initialSupplies = [
  {
    id: 1, brand: 'Dell', serial: 'DELL123456', status: 'Available',
  },
  {
    id: 2, brand: 'HP', serial: 'HP789012', status: 'In use',
  },
  {
    id: 3, brand: 'Lenovo', serial: 'LEN345678', status: 'In storage',
  },
  {
    id: 4, brand: 'Apple', serial: 'APP901234', status: 'Available',
  },
  {
    id: 5, brand: 'Asus', serial: 'ASUS567890', status: 'Stolen',
  },
];

const supplyStatuses = ['Available', 'In storage', 'In use', 'Stolen'];

interface SupplyContextProps {
  status: typeof supplyStatuses;
  supplies: typeof initialSupplies;
  filter: string;
  currentPage: number;
  suppliesPerPage: number;
  addSupply: () => void;
  editSupply: (id: number) => void;
  deleteSupply: () => void;
  confirmDelete: (supply: any) => void;
  confirmadd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToDelete: any;
  setSupplyToDelete: React.Dispatch<React.SetStateAction<any>>;
  addModal: boolean;
  setaddModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToadd: any;
  setSupplyToadd: React.Dispatch<React.SetStateAction<any>>;
}

const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderComputer({
  children,
}: {
  children: ReactNode;
}) {
  const [supplies, setSupplies] = useState(initialSupplies);
  const [filter, setFilter] = useState('');
  const [currentPage, setCurrentPage] = useState(1);
  const [suppliesPerPage] = useState(5);
  const [deleteModal, setDeleteModal] = useState(false);
  const [addModal, setaddModal] = useState(false);
  const [supplyToDelete, setSupplyToDelete] = useState(null);
  const [supplyToadd, setSupplyToadd] = useState(null);
  const [status] = useState(supplyStatuses);

  // Simulate adding a supply
  const addSupply = () => {
    console.log('Add supply');
  };

  // Simulate editing a supply
  const editSupply = (id: number) => {
    console.log('Edit supply', id);
  };

  // Open confirmation modal for deletion
  const confirmDelete = (supply: any) => {
    setSupplyToDelete(supply);
    setDeleteModal(true);
  };

  const confirmadd = () => {
    setaddModal(true);
  };

  // Delete supply
  const deleteSupply = () => {
    if (supplyToDelete) {
      setSupplies(supplies.filter((supply) => supply.id !== supplyToDelete.id));
      setDeleteModal(false);
      setSupplyToDelete(null);
    }
  };

  // Simulate status change
  const changeStatus = (id: number, newStatus: string) => {
    setSupplies(
      supplies.map((supply) => (supply.id === id ? { ...supply, status: newStatus } : supply)),
    );
  };

  return (
    <SupplyContext.Provider
      value={{
        status,
        supplies,
        filter,
        currentPage,
        suppliesPerPage,
        addSupply,
        editSupply,
        deleteSupply,
        confirmDelete,
        confirmadd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        supplyToDelete,
        setSupplyToDelete,
        addModal,
        setaddModal,
        supplyToadd,
        setSupplyToadd,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}

export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
