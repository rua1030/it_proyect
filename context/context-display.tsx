'use client';

import React, {
  createContext, useState, useContext, ReactNode,
} from 'react';

// Simulated supply data
const initialSupplies = [
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
  {
    id: 1,
    brand: 'Dell',
    serial: 'DELL123456',
    type: 'Laptop',
    status: 'In use',
  },
  {
    id: 2,
    brand: 'HP',
    serial: 'HP789012',
    type: 'Printer',
    status: 'Available',
  },
  {
    id: 3,
    brand: 'Lenovo',
    serial: 'LEN345678',
    type: 'Desktop',
    status: 'In maintenance',
  },
  {
    id: 4,
    brand: 'Apple',
    serial: 'APP901234',
    type: 'Tablet',
    status: 'In use',
  },
  {
    id: 5,
    brand: 'Asus',
    serial: 'ASUS567890',
    type: 'Monitor',
    status: 'Available',
  },
];

const supplyStatuses = ['In use', 'In maintenance', 'Available'];

// Define the shape of the context
interface SupplyContextProps {
  supplyStatuse: typeof supplyStatuses;
  supplies: typeof initialSupplies;
  filter: string;
  currentPage: number;
  suppliesPerPage: number;
  addSupply: () => void;
  editSupply: (id: number) => void;
  deleteSupply: () => void;
  confirmDelete: (supply: string) => void;
  confirmAdd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToDelete: string;
  setSupplyToDelete: React.Dispatch<React.SetStateAction<string>>;
  addModal: boolean;
  setAddModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToAdd: string;
  setSupplyToAdd: React.Dispatch<React.SetStateAction<string>>;
}

const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderDevices({ children }: { children: ReactNode }) {
  const [supplyStatuse] = useState(supplyStatuses);
  const [supplies, setSupplies] = useState(initialSupplies);
  const [filter, setFilter] = useState('');
  const [currentPage] = useState(1);
  const [suppliesPerPage] = useState(5);
  const [deleteModal, setDeleteModal] = useState(false);
  const [supplyToDelete, setSupplyToDelete] = useState(null);
  const [addModal, setAddModal] = useState(false);
  const [supplyToAdd, setSupplyToAdd] = useState(null);

  const addSupply = () => {
    console.log('Add supply');
  };

  // Simulate editing a supply
  const editSupply = (id: number) => {
    console.log('Edit supply', id);
  };

  // Open confirmation modal for deletion
  const confirmDelete = (supply: string) => {
    setSupplyToDelete(supply);
    setDeleteModal(true);
  };

  const confirmAdd = () => {
    setAddModal(true);
  };

  // Delete supply
  const deleteSupply = () => {
    if (supplyToDelete) {
      setSupplies(supplies.filter((supply) => supply.id !== supplyToDelete.id));
      setDeleteModal(false);
      setSupplyToDelete(null);
    }
  };

  // Simulate status change
  const changeStatus = (id: number, newStatus: string) => {
    setSupplies(
      supplies.map((supply) => (supply.id === id ? { ...supply, status: newStatus } : supply)),
    );
  };

  return (
    <SupplyContext.Provider
      value={{
        supplyStatuse,
        supplies,
        filter,
        currentPage,
        suppliesPerPage,
        addSupply,
        editSupply,
        deleteSupply,
        confirmDelete,
        confirmAdd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        supplyToDelete,
        setSupplyToDelete,
        addModal,
        setAddModal,
        supplyToAdd,
        setSupplyToAdd,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}

export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
