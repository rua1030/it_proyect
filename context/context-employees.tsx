'use client';

import React, {
  createContext, useState, useContext, ReactNode,
  useEffect,
} from 'react';
import { EmployeesService } from '@/services/employeesService';
import { Employees } from '@/models/employees';

// interface Employee {
//   id: number; // remover
//   name: string;
//   email: string;
//   idNumber: string;
//   jobTitle: string;
//   operatingSystem: string;
//   status: string;
// }

// const initialEmployees: Employees[] = [
//   {
//     id: '1',
//     name: 'John Doe',
//     email: 'john@example.com',
//     idNumber: '1234567890',
//     jobTitle: 'Developer',
//     operatingSystem: 'Windows',
//     status: 'Onbording',
//   },
//   {
//     id: '2',
//     name: 'Jane Smith',
//     email: 'jane@example.com',
//     idNumber: '0987654321',
//     jobTitle: 'Designer',
//     operatingSystem: 'macOS',
//     status: 'Active',
//   },
//   {
//     id: '3',
//     name: 'Bob Johnson',
//     email: 'bob@example.com',
//     idNumber: '2345678901',
//     jobTitle: 'Manager',
//     operatingSystem: 'Linux',
//     status: 'Offbording',
//   },
//   {
//     id: '4',
//     name: 'Alice Brown',
//     email: 'alice@example.com',
//     idNumber: '3456789012',
//     jobTitle: 'HR Specialist',
//     operatingSystem: 'Windows',
//     status: 'Active',
//   },
//   {
//     id: '5',
//     name: 'Charlie Wilson',
//     email: 'charlie@example.com',
//     idNumber: '4567890123',
//     jobTitle: 'Tester',
//     operatingSystem: 'macOS',
//     status: 'Onbording',
//   },
//   {
//     id: '6',
//     name: 'John Doe 2',
//     email: 'john@example.com',
//     idNumber: '1234567890',
//     jobTitle: 'Developer',
//     operatingSystem: 'Windows',
//     status: 'Onbording',
//   },
// ];

const employeesService = new EmployeesService();

const employeeStatuses = ['Onbording', 'Active', 'Offboarding'];
const operatingSystems = ['Windows', 'macOS', 'Linux'];

interface SupplyContextProps {
  statusSystems: typeof operatingSystems;
  status: typeof employeeStatuses;
  employees: typeof EmployeesService;
  filter: string;
  addEmployee: () => void;
  editEmployee: (id: number) => void;
  deleteEmployee: () => void;
  confirmDelete: (employee: Employees) => void;
  modalAdd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  changeOS: (id: number, newOS: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  employeeToDelete: Employees | null;
  setEmployeeToDelete: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  addModal: boolean;
  setAddModal: React.Dispatch<React.SetStateAction<boolean>>;

  changePage: (pageNumber: number) => void;
  currentPage: number;

  currentEmployees: Employees[];
  filteredEmployees: Employees[];

  employeesPerPage: number;
}
const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderEmployees({ children }: { children: ReactNode }) {
  const [status] = useState(employeeStatuses);
  const [statusSystems] = useState(operatingSystems);
  const [employees, setEmployees] = useState<Employees[]>([]);
  const [filter, setFilter] = useState('');
  const [deleteModal, setDeleteModal] = useState(false);
  const [addModal, setAddModal] = useState(false);
  const [employeeToDelete, setEmployeeToDelete] = useState(null);

  const [currentPage, setCurrentPage] = useState(1);
  const [employeesPerPage] = useState(5);

  const fetchEmployees = async () => {
    try {
      const fetchedEmployees = await employeesService.getAllEmployees();
      setEmployees(fetchedEmployees);
    } catch (error) {
      console.error('Error fetching Employees:', error);
    }
  }; 

  // Simulate accion an employee
  const addEmployee = () => {
    console.log('Add employee');
  };

  const editEmployee = (id: number) => {
    console.log('Edit employee', id);
  };

  const confirmDelete = (employee: string) => {
    setEmployeeToDelete(employee);
    setDeleteModal(true);
  };

  const modalAdd = () => {
    setAddModal(true);
  };
  // const setAddEmployees = () => {
  //   console.log('Add employee');
  //   setAddEmployees();
  //   setAddModal(false);
  // };

  const deleteEmployee = () => {
    if (employeeToDelete) {
      setEmployees(
        employees.filter((employee) => employee.id !== employeeToDelete.id),
      );
      setDeleteModal(false);
      setEmployeeToDelete(null);
    }
  };

  const changeStatus = (id: number, newStatus: string) => {
    setEmployees(
      employees.map((employee) => (employee.id === id
        ? { ...employee, status: newStatus } : employee)),
    );
  };

  const changeOS = (id: number, newOS: string) => {
    setEmployees(
      employees.map((employee) => (employee.id === id
        ? { ...employee, operatingSystem: newOS } : employee)),
    );
  };

  const filteredEmployees = employees.filter(
    (employee) => employee.name.toLowerCase().includes(filter.toLowerCase())
      || employee.email.toLowerCase().includes(filter.toLowerCase())
      || employee.idNumber.includes(filter)
      || employee.jobTitle.toLowerCase().includes(filter.toLowerCase())
      || employee.operatingSystem.toLowerCase().includes(filter.toLowerCase()),
  );

  const indexOfLastEmployee = currentPage * employeesPerPage;
  const indexOfFirstEmployee = indexOfLastEmployee - employeesPerPage;
  const currentEmployees = filteredEmployees.slice(
    indexOfFirstEmployee,
    indexOfLastEmployee,
  );

  const changePage = (pageNumber: number) => setCurrentPage(pageNumber);

  useEffect(() => {
    fetchEmployees();
  }, []);

  return (
    <SupplyContext.Provider
      value={{
        employees,
        filter,
        addEmployee,
        editEmployee,
        deleteEmployee,
        confirmDelete,
        modalAdd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        addModal,
        setAddModal,
        changeOS,
        statusSystems,
        status,
        employeeToDelete,
        setEmployeeToDelete,
        changePage,
        currentPage,
        currentEmployees,
        filteredEmployees,
        employeesPerPage,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}
export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
