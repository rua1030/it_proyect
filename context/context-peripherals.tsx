'use client';

import React, {
  createContext, useState, useContext, ReactNode,
} from 'react';

const initialPeripherals = [
  {
    id: 1,
    brand: 'Logitech',
    name: 'MX Master 3',
    quantity: 50,
    status: 'In stock',
  },
  {
    id: 2,
    brand: 'Dell',
    name: 'P2419H Monitor',
    quantity: 30,
    status: 'Low stock',
  },
  {
    id: 3,
    brand: 'Corsair',
    name: 'K70 RGB Keyboard',
    quantity: 20,
    status: 'Out of stock',
  },
  {
    id: 4,
    brand: 'Jabra',
    name: 'Evolve 75 Headset',
    quantity: 40,
    status: 'In stock',
  },
  {
    id: 5,
    brand: 'Microsoft',
    name: 'Surface Dock',
    quantity: 15,
    status: 'Low stock',
  },
];

const peripheralStatuses = ['In stock', 'Low stock', 'Out of stock'];

interface SupplyContextProps {
  statuses: typeof peripheralStatuses;
  peripherals: typeof initialPeripherals;
  filter: string;
  currentPage: number;
  addPeripheral: () => void;
  editPeripheral: (id: number) => void;
  deletePeripheral: () => void;
  confirmDelete: (supply: string) => void;
  confirmadd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  peripheralToDelete: string;
  setPeripheralToDelete: React.Dispatch<React.SetStateAction<string>>;
  addModal: boolean;
  setaddModal: React.Dispatch<React.SetStateAction<boolean>>;
  peripheralToadd: string;
  setPeripheralToadd: React.Dispatch<React.SetStateAction<string>>;
}

const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderPeripherals({
  children,
}: {
  children: ReactNode;
}) {
  const [peripherals, setPeripherals] = useState(initialPeripherals);
  const [filter, setFilter] = useState('');
  const [currentPage] = useState(1);
  const [deleteModal, setDeleteModal] = useState(false);
  const [peripheralToDelete, setPeripheralToDelete] = useState(null);
  const [addModal, setaddModal] = useState(false);
  const [peripheralToadd, setPeripheralToadd] = useState(null);
  const [statuses] = useState(peripheralStatuses);

  const addPeripheral = () => {
    console.log('Add peripheral');
  };

  // Simulate editing a peripheral
  const editPeripheral = (id: number) => {
    console.log('Edit peripheral', id);
  };

  // Open confirmation modal for deletion
  const confirmDelete = () => {
    setDeleteModal(true);
  };

  const confirmadd = () => {
    setaddModal(true);
  };

  // Delete peripheral
  const deletePeripheral = () => {
    if (peripheralToDelete) {
      setPeripherals(
        peripherals.filter(
          (peripheral) => peripheral.id !== peripheralToDelete.id,
        ),
      );
      setDeleteModal(false);
      setPeripheralToDelete(null);
    }
  };

  // Simulate status change
  const changeStatus = (id: number, newStatus: string) => {
    setPeripherals(
      peripherals.map((peripheral) => (peripheral.id === id ? { ...peripheral, status: newStatus } : peripheral)),
    );
  };
  return (
    <SupplyContext.Provider
      value={{
        peripherals,
        filter,
        currentPage,
        addPeripheral,
        editPeripheral,
        deletePeripheral,
        confirmDelete,
        confirmadd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        peripheralToDelete,
        setPeripheralToDelete,
        addModal,
        setaddModal,
        peripheralToadd,
        setPeripheralToadd,
        statuses,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}
export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
