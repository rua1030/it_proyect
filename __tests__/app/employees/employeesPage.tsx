import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { SupplyProviderEmployees } from '@/context/context-employees';
import EmployeeManagement from '@/app/employees/page';

describe('EmployeeManagement', () => {
  const setup = () => {
    render(
      <SupplyProviderEmployees>
        <EmployeeManagement />
      </SupplyProviderEmployees>,
    );
  };

  it('matches snapshot', () => {
    const { asFragment } = render(
      <SupplyProviderEmployees>
        <EmployeeManagement />
      </SupplyProviderEmployees>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all necessary fields', () => {
    setup();
    expect(screen.getByText('Employee Management')).toBeInTheDocument();
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('Email')).toBeInTheDocument();
    expect(screen.getByText('ID Number')).toBeInTheDocument();
    expect(screen.getByText('Position')).toBeInTheDocument();
    expect(screen.getByText('Operating System')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Actions')).toBeInTheDocument();
    expect(screen.getByText('Previous')).toBeInTheDocument();
    expect(screen.getByText('Next')).toBeInTheDocument();
  });

  it('renders Employee Management title', () => {
    setup();
    expect(screen.getByText('Employee Management')).toBeInTheDocument();
  });

  it('updates filter on input change', () => {
    setup();
    const input = screen.getByPlaceholderText(
      'Search by name, email, or ID number...',
    );
    fireEvent.change(input, { target: { value: 'jhon' } });
    expect(input).toHaveValue('jhon');
  });

  it('opens add employee modal on button click', () => {
    setup();
    const addButton = screen.getByText('Add Employeed');
    fireEvent.click(addButton);
    expect(screen.getByText('Add employee')).toBeInTheDocument();
  });

  it('changes page when pagination button is clicked', () => {
    setup();
    const nextPageButton = screen.getByText(/Next/i);
    fireEvent.click(nextPageButton);
  });
});
