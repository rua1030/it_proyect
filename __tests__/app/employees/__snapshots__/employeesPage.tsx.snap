// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeManagement matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto p-4"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      Employee Management
    </h1>
    <div
      class="flex justify-between items-center mb-4"
    >
      <input
        class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 max-w-sm"
        placeholder="Search by name, email, or ID number..."
        type="text"
        value=""
      />
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      >
        <svg
          class="lucide lucide-plus mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Add Employeed
      </button>
    </div>
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Name
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Email
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              ID Number
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Position
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Operating System
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Status
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0"
        />
      </table>
    </div>
    <div
      class="flex justify-center mt-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
        disabled=""
      >
        Previous
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mx-1"
      >
        1
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 ml-2"
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
